public/app/plugins/datasource/loki/querySplitting.ts

```diff
@@ -179,19 +173,45 @@ export function runPartitionedQueries(datasource: LokiDatasource, request: DataQ
   const [instantQueries, normalQueries] = partition(queries, (query) => query.queryType === LokiQueryType.Instant);
   const [logQueries, metricQueries] = partition(normalQueries, (query) => isLogsQuery(query.expr));
 
+  const oneDayMs = 24 * 60 * 60 * 1000;
+  const rangePartitionedLogQueries = groupBy(logQueries, (query) =>
+    query.experimentalChunkRange && isValidDuration(query.experimentalChunkRange)
+      ? durationToMilliseconds(parseDuration(query.experimentalChunkRange!))
+      : oneDayMs
+  );
+  const rangePartitionedMetricQueries = groupBy(metricQueries, (query) =>
+    query.experimentalChunkRange && isValidDuration(query.experimentalChunkRange)
+      ? durationToMilliseconds(parseDuration(query.experimentalChunkRange!))
+      : oneDayMs
+  );
+
   const requests: LokiGroupedRequest = [];
-  if (logQueries.length) {
+  for (const [chunkRangeMs, queries] of Object.entries(rangePartitionedLogQueries)) {
```

Comment:
Not really in love of this part.

```suggestion
  for (const chunkRangeMs in rangePartitionedLogQueries) {
```

Being queries: `rangePartitionedLogQueries[chunkRangeMs]`.

Reply:
Could you elaborate why you prefer that?

I don't like `for .. in` because by that you would also iterate over "inherited" properties: https://jsfiddle.net/w1dsunke/

Reply:
Oh interesting, I think `[chunkRangeMs, queries] of Object.entries()` is much more readable.

In this case there are no inherited properties, yes. But that was the reason why I generally don't tend to use `for .. in`.